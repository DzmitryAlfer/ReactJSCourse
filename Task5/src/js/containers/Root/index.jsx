import React, { Component } from 'react'
import { Provider } from 'react-redux'
import PropTypes from "prop-types";
import {DocumentComponent} from "../../components/DocumentComponent";
/*import { Router, RouterContext } from 'react-router'*/

export default class Root extends Component {
    render() {
        const { store, history, routes, type, renderProps } = this.props

        return (
            <Provider store={store}>
                <DocumentComponent/>
            </Provider>
            /*<Provider store={store}>
                <div>
                    {{ type === 'server'
                        ? <RouterContext {...renderProps} />
                        : <Router history={history} routes={routes} />
                    }
                    <DevTools />}
                </div>
            </Provider>*/
        )
    }
}

Root.propTypes = {
    store: PropTypes.object.isRequired,
    /*history: PropTypes.object.isRequired,
    routes: PropTypes.node.isRequired*/
};