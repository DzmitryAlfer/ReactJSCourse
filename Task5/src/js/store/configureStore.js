import { createStore, applyMiddleware, compose } from 'redux'
import { createLogger } from 'redux-logger'
import createSagaMiddleware, { END } from 'redux-saga'
import rootReducer from '../reducers'
import { persistStore, persistReducer } from 'redux-persist'
import storage from 'redux-persist/lib/storage'

const persistConfig = {
    key: 'root',
    storage,
};

const persistedReducer = persistReducer(persistConfig, rootReducer)

export default function configureStore(initialState) {
    const sagaMiddleware = createSagaMiddleware()

    const store = createStore(
        persistedReducer,
        initialState,
        compose(
            applyMiddleware(
                sagaMiddleware,
                createLogger()
            )
        )
    );
    const persistor = persistStore(store);

    /*if (module.hot) {
        // Enable Webpack hot module replacement for reducers
        module.hot.accept('../reducers', () => {
            const nextRootReducer = require('../reducers').default;
            store.replaceReducer(nextRootReducer);
        });
    }*/

    store.runSaga = sagaMiddleware.run;
    store.close = () => store.dispatch(END);
    return {store, persistor};
}