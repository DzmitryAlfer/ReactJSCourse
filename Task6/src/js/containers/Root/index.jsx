import React, { Component } from 'react'
import { Provider } from 'react-redux'
import PropTypes from "prop-types";
import { PersistGate } from 'redux-persist/integration/react'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
import {withHomePageTemplate} from '../../components/pages/HomePageTemplate'
import {EmptyResultStatePage} from "../../components/pages/EmptyResultStatePage";
import {Page404} from "../../components/pages/Page404";
import {SearchResultsPage} from "../../components/pages/SearchResultsPage";

export default class Root extends Component {
    render() {
        const { store, persistor} = this.props;

        return (
            <Provider store={store}>
                <PersistGate loading={null} persistor={persistor}>
                    <Router>
                        <Switch>
                            <Route exact path='/' component={withHomePageTemplate(EmptyResultStatePage)}/>
                            {/*<Route exact path='/search' component={withHomePageTemplate(SearchResultsPage)}/>*/}
                            <Route path='/search/:searchQuery?' component={withHomePageTemplate(SearchResultsPage)}/>
                            <Route path='*' component={Page404}/>
                        </Switch>
                    </Router>
                </PersistGate>
            </Provider>
        )
    }
}

Root.propTypes = {
    store: PropTypes.object.isRequired,
    /*history: PropTypes.object.isRequired,
    routes: PropTypes.node.isRequired*/
};