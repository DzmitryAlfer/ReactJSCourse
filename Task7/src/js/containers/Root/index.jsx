import React, { Component } from 'react'
import { Provider } from 'react-redux'
import PropTypes from "prop-types";
import { PersistGate } from 'redux-persist/integration/react'
import { Route, Switch } from 'react-router-dom'
import {withHomePageTemplate} from '../../components/pages/HomePageTemplate'
import {EmptyResultStatePage} from "../../components/pages/EmptyResultStatePage";
import {Page404} from "../../components/pages/Page404";
import {SearchResultsPage} from "../../components/pages/SearchResultsPage";
import {FilmDescriptionPage} from "../../components/pages/FilmDescriptionPage";

export default class Root extends Component {
    render() {
        const { Router, store, persistor, requestLocation, context} = this.props;

        return (
           /*<Provider store={store}>
                <PersistGate loading={null} persistor={persistor}>
                    <Router context={context} location={requestLocation}>
                        <Switch>
                            <Route exact path='/' component={withHomePageTemplate(EmptyResultStatePage)}/>
                            <Route path='/search/:searchQuery?' component={withHomePageTemplate(SearchResultsPage)}/>
                            <Route path='/film/:movieId' component={FilmDescriptionPage}/>
                            <Route path='*' component={Page404}/>
                        </Switch>
                    </Router>
                </PersistGate>
            </Provider>*/

            <Provider store={store}>
                    <Router context={context} location={requestLocation}>
                        <Switch>
                            <Route exact path='/' component={withHomePageTemplate(EmptyResultStatePage)}/>
                            <Route path='/search/:searchQuery?' component={withHomePageTemplate(SearchResultsPage)}/>
                            <Route path='/film/:movieId' component={FilmDescriptionPage}/>
                            <Route path='*' component={Page404}/>
                        </Switch>
                    </Router>
            </Provider>
        )
    }
}

Root.propTypes = {
    store: PropTypes.object,
    context: PropTypes.object,
    Router: PropTypes.func,
    location: PropTypes.string,
};