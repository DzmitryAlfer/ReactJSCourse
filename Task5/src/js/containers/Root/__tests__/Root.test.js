import React from 'react';
import Renderer from 'react-test-renderer';
import Root from "../";

import configureStore from "../../../store/configureStore";

const store = configureStore();

jest.mock('../../../components/DocumentComponent');

describe('Root', () => {
    test('Component render', () => {
        const tree = Renderer.create(
            <Root store={store}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});