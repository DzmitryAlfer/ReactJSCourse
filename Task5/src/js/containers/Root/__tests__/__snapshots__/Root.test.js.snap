// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root Component render 1`] = `
<span>
  DocumentComponent MOCK
</span>
`;
